ci/tests/backend/alba.py

```diff
@@ -232,21 +238,96 @@ def is_bucket_count_valid_with_policy(bucket_count, policies):
-            print policy
             pol_k, pol_m, pol_c, pol_x = tuple(policy)
-            print bc_policy, tuple(bc_policy)
             bc_k, bc_m, bc_c, bc_x = tuple(bc_policy)
             safe = (pol_k == bc_k) and (pol_m == bc_m) and (bc_c >= pol_c) and (bc_x <= pol_c)
 
     return safe
 
 
-def claim_disks(alba_backend, nr_of_disks, disk_type='sata'):
-    pass
+def get_all_disks(alba_backend_guid):
+    api = Connection.get_connection()
+    alba_backend = api.fetch('alba/backends', alba_backend_guid)
+    if 'all_disks' in alba_backend:
+        all_disks = alba_backend['all_disks']
+    else:
+        all_disks = list()
+
+    return all_disks
+
+
+def filter_disks(available_disks, amount, disk_type=''):
+    hdds, ssds = get_physical_disks(GRID_IP)
+    count = 0
+    filtered_disks = list()
+
+    if disk_type == 'SATA':
+        list_to_check = hdds.values()
+    elif disk_type == 'SSD':
+        list_to_check = ssds.values()
+    else:
+        hdds.update(ssds)
+        list_to_check = hdds.values()
+
+    for disk_name in available_disks:
+        for disk in list_to_check:
+            if disk_name == disk['name']:
+                filtered_disks.append(disk['name'])
+                count += 1
+        if count == amount:
+            break
+
+    return filtered_disks
+
+
+def initialise_disks(alba_backend_guid, nr_of_disks, disk_type=''):
+    # assume no disks are claimed by a remote environment
+    all_disks = get_all_disks(alba_backend_guid)
+
+    initialised_disks = [disk['name'] for disk in all_disks if disk['status'] == 'available']
+    nr_of_disks_to_init = nr_of_disks - len(initialised_disks)
+    if nr_of_disks_to_init <= 0:
+        return True
+
+    uninitialized_disks = [disk['name'] for disk in all_disks if disk['status'] == 'uninitialized']
+    assert len(uninitialized_disks) >= nr_of_disks_to_init, "Not enough disks to initialize!"
+
+    disks_to_init = filter_disks(uninitialized_disks, nr_of_disks_to_init, disk_type)
+    assert len(disks_to_init) >= nr_of_disks_to_init, "Not enough disks to initialize!"
+
+    alba_node = AlbaNodeList.get_albanode_by_ip(GRID_IP)
+    failures = AlbaNodeController.initialize_disks(alba_node.guid, disks_to_init)
+    assert not failures, 'Alba disk initialization failed for (some) disks: {0}'.format(failures)
+
+
+def claim_disks(alba_backend_guid, nr_of_disks, disk_type=''):
+    all_disks = get_all_disks(alba_backend_guid)
+
+    claimed_disks = [disk['name'] for disk in all_disks if 'status' in disk and disk['status'] == 'claimed' and 'name' in disk]
+    nr_disks_to_claim = nr_of_disks - len(claimed_disks)
+    if nr_disks_to_claim <= 0:
+        return True
+
+    initialise_disks(alba_backend_guid, nr_disks_to_claim, disk_type)
+
+    all_disks = get_all_disks(alba_backend_guid)
+    osds = dict()
+
+    claimable_disks = [disk['name'] for disk in all_disks if 'status' in disk and disk['status'] == 'available' and 'asd_id' in disk]
```

Comment:
How long does this code wait? It can take a few minutes for disks to get "available"